Je n'arrive pas à compiler le code sur macOS

Bonjour ! 

Je n'ai jamais codé en C++ (pas depuis 25 ans en tout cas !) et je n'arrive pas à compiler ce code sur **macOS**, que ce soit avec `c++` ou `g++` je me retrouve avec l'erreur suivante:

```shell
# g++ Application.cpp
Application.cpp:320:44: error: expected expression
        for_each(line.begin(), line.end(), [](char & c)
                                           ^
1 error generated.
```

Pouvez-vous me débloquer ?

Bonjour David et @Farzy,

Effectivement, chez moi ça ne compile pas avec mon compilateur habituel. J'ai essayé avec les versions suivantes :

- g++ 10.2.1-1 (debian bullseye)
- clang 11.0-51+nmu5 (debian bullseye)
- clang 13.0.1-3 (debian sid)

Dans les trois cas j'ai une erreur, voici ce que ça donne avec la version `clang 10.2.1-1` :

```bash
~/tmp/WordleSutom$ clang Application.cpp 
Application.cpp:320:9: error: use of undeclared identifier 'for_each'
        for_each(line.begin(), line.end(), [](char & c)
        ^
Application.cpp:325:12: error: no matching function for call to 'find'
        if(find(words.begin(), words.end(), line) == words.end() )
           ^~~~
/usr/bin/../lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/bits/streambuf_iterator.h:421:5: note: candidate template ignored: could no
t match 'istreambuf_iterator' against '__normal_iterator'
    find(istreambuf_iterator<_CharT> __first,
    ^
2 errors generated.
```

J'ai exactement la même sortie avec clang `11.0-51`.

Quelle version de clang utilises-tu ?
Bonne journée,
Le Barde.

--- a/Application.cpp
+++ b/Application.cpp
@@ -1,4 +1,5 @@
 #include <cmath>
+#include <algorithm>
 #include <iostream>
 #include <fstream>
 #include <random>

